### PR TITLE
Feature/omni examples

### PR DESCRIPTION
**Pull request by dweng0**

### CODE REVIEW

**Comment by Gulikovskiy** [comment]
Screenshot 2023-02-13 at 16 47 57

A few small things(DEPOSITING):
1. no symbol
2. POI balance was not updated after a tx
3. wrong symbol

**Comment by Gulikovskiy** [comment]
Screenshot 2023-02-13 at 16 54 36

REDEEM:
1. no symbol
2. POI balance was not updated after a tx

File: examples/omni-router/src/prepare.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  const reserveToken = new Erc20(account, omniRouter.address)
+  const token = tokenAddress ? new Erc20(account, tokenAddress) : undefined
+
+  const amount = getEnv('AMOUNT')
+
+  const createErc20 = async (address: Address) =>
+    await new Erc20(account, address)
+
+  return {
```
Comment (Gulikovskiy) [inline examples/omni-router/src/prepare.ts:45]: you don't use `provider`, `createErc20` and `reserveToken`. I think we can get rid of unused variables

File: examples/omni-router/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    )
+    const preTokens = await token?.contract.balanceOf(
+      await index.account.address(),
+    )
+
+    console.log('current balance:')
+    console.dir(`${utils.formatEther(preBalance)}`)
+
+    console.log('current tokens: ')
+    console.dir(`${utils.formatUnits(preTokens || '0', 6)} USDC`)
```
Comment (Gulikovskiy) [inline examples/omni-router/src/index.ts:23]: 6 => use `token.decimals()`
USDC => use `token.symbol()`

File: examples/omni-router/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  }
+
+  if (isDeposit) {
+    await getBeforeData()
+    const tokenAmount = utils.parseUnits(amount, 6) //TODO
```
Comment (Gulikovskiy) [inline examples/omni-router/src/index.ts:28]: 6 => `token.decimals()`

File: examples/omni-router/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+      )
+      const postTokens =
+        (await token?.contract.balanceOf(await index.account.address())) ||
+        BigNumber.from(0)
+
+      console.log('Balances: ')
+      console.dir(`${utils.formatEther(postBalance)} ${await index.symbol()}`)
+
+      console.log('Tokens:')
+      console.dir(`${utils.formatUnits(postTokens, 6)} ${await index.symbol()}`)
```
Comment (Gulikovskiy) [inline examples/omni-router/src/index.ts:55]: the same here

Comment (Gulikovskiy) [reply examples/omni-router/src/index.ts:55]: + wrong symbol :)

File: examples/omni-router/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    }
+  } else {
+    await getBeforeData()
+
+    const indexAmount = utils.parseUnits(amount, 18)
```
Comment (Gulikovskiy) [inline examples/omni-router/src/index.ts:60]: 18 => `index.decimals()`

**Comment by dweng0** [comment]
image

File: examples/omni-router/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+  } else {
+    await getBeforeData()
+
+    const indexAmount = utils.parseUnits(amount, indexDecimals)
+    console.log('Redeem amount:')
+    console.dir(`${utils.formatEther(indexAmount)} ${indexSymbol}`)
+
+    const previewInfo = await omniRouter.previewRedeem(indexAmount)
+    console.log('(Preview) Redeem Amount:')
+    console.dir(`${utils.formatUnits(previewInfo || '0', 6)} ${tokenSymbol}`)
```
Comment (Gulikovskiy) [inline examples/omni-router/src/index.ts:75]: 6 => `tokenDecimals`

File: examples/omni-router/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+        indexAddress,
+      )
+      const postTokens =
+        (await token?.contract.balanceOf(indexAddress)) || BigNumber.from(0)
+
+      console.log('Balances: ')
+      console.dir(`${utils.formatEther(postRedeemBalance)} ${indexSymbol}`)
+
+      console.log('Tokens:')
+      console.dir(`${utils.formatUnits(postTokens || '0', 6)} ${tokenSymbol}`)
```
Comment (Gulikovskiy) [inline examples/omni-router/src/index.ts:100]: 6 => `tokenDecimals`

**Review by Gulikovskiy — APPROVED** [review]

**Pull request merged**